### PR TITLE
Chore/locking unlocking data

### PR DESCRIPTION
**Pull request by noumanahmads**
## Description

Expose locking/unlocking data

1. Create a table for saving the information for locked tokens and their unlocking block
2. Updated the `v1/import` script for inserting the data in the table
3. Updated the APIs for returning the expected response and their schema files.
4. Tests updated

For details refer to issue #515

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Testing information
New tests are written and also updated tests.

## Checklist
- [x] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [x] `npm run test` passes
- [ ] Changelog is updated
- [x] @zone117x for review

### CODE REVIEW

File: src/datastore/postgres-store.ts [hunk 1]
```diff
@@ -3756,6 +3761,71 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
...
+  }
+
+  async getTokenOfferingLocked(address: string) {
+    return this.query(async client => {
+      const queryResult = await client.query<DbTokenOfferingLocked & { total: BigInt }>(
+        `
+         SELECT block, value, (SUM(value) OVER())::bigint AS total
+         FROM token_offering_locked
+         WHERE address = $1
+         GROUP BY id;
```
Comment (zone117x) [inline src/datastore/postgres-store.ts:3801]: What is the purpose of the `GROUP By id` statement?

Comment (noumanahmads) [reply src/datastore/postgres-store.ts:3801]: Initially I was using this query 
```
SELECT id, block, value, SUM(value)::bigint AS total
FROM token_offering_locked
WHERE address = $1
``` 
and it was giving me this error
`error: column "token_offering_locked.id" must appear in the GROUP BY clause or be used in an aggregate function`
So I added` GROUP BY`

But with this sum function `(SUM(value) OVER())::bigint AS total`, I think this `GROUP BY id` should be removed.

**Review by zone117x — COMMENTED** [review]
This looks great! @noumanahmads is there anything you'd change in this PR to help with the situation outlined at https://github.com/blockstack/stacks-blockchain-api/issues/393#issuecomment-820491702 ?

If not, we can merge this PR as is, but we can't fully close https://github.com/blockstack/stacks-blockchain-api/issues/515 until #393 is resolved. 

EDIT: testing this PR with my WIP for #393 and it's working well.

**Review by zone117x — APPROVED** [review]

File: src/datastore/postgres-store.ts [hunk 2]
```diff
@@ -3762,6 +3767,70 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
...
+      columnCount,
+    });
+    const values: any[] = [];
+    for (const lockedInfo of lockedInfos) {
+      values.push(lockedInfo.address, lockedInfo.value, lockedInfo.block);
+    }
+    const insertQuery = `INSERT INTO token_offering_locked (
+      address, value, block
+      ) VALUES ${insertParams}`;
+    const insertQueryName = `insert-batch-subdomains_${columnCount}x${lockedInfos.length}`;
```
Comment (zone117x) [inline src/datastore/postgres-store.ts:3783]: Just saw this, the query name can't be a duplicated (`insert-batch-subdomains`)

File: src/import-v1/index.ts [hunk 3]
```diff
@@ -333,6 +340,70 @@ async function* readSubdomains(importDir: string) {
...
+  _transform(data: string, _encoding: string, callback: stream.TransformCallback) {
+    if (data.startsWith('-----END')) {
+      this.isVesting = false;
+    }
+
+    if (this.isVesting) {
+      const parts = data.split(',');
+      if (parts[0] !== 'address') {
+        const tokenOfferingLocked: DbTokenOfferingLocked = {
+          address: parts[0],
```
Comment (zone117x) [inline src/import-v1/index.ts:358]: This `address` string can be a b58 (bitcoin) address or a STX address. It needs to be converted to a STX address if it's b58. Similar to what was done here: https://github.com/blockstack/stacks-blockchain-api/pull/542

**Review by zone117x — CHANGES_REQUESTED** [review]

File: src/datastore/postgres-store.ts (hunk 2 above)
Comment (noumanahmads) [reply src/datastore/postgres-store.ts]: fixed this one.

File: src/import-v1/index.ts (hunk 3 above)
Comment (noumanahmads) [reply src/import-v1/index.ts]: Updated this address part.

**Comment by noumanahmads** [comment]
> #393

I think you have updated that part in this [PR#556](https://github.com/blockstack/stacks-blockchain-api/pull/556). Do we need to add anything else?

File: src/tests/datastore-tests.ts [hunk 4]
```diff
@@ -18,11 +18,14 @@ import {
...
   DbBnsName,
   DbBnsSubdomain,
+  DbTokenOfferingLocked,
 } from '../datastore/common';
 import { PgDataStore, cycleMigrations, runMigrations } from '../datastore/postgres-store';
 import { PoolClient } from 'pg';
 import { parseDbEvent } from '../api/controllers/db-controller';
 import * as assert from 'assert';
+import { importV1 } from '../import-v1';
+import supertest = require('supertest');
```
Comment (zone117x) [inline src/tests/datastore-tests.ts:28]: Accidental imports?

**Pull request merged**